### PR TITLE
Fix fabcar Java chaincode test & coverage failures

### PR DESCRIPTION
**Pull request by jt-nti**
Signed-off-by: James Taylor <jamest@uk.ibm.com>

### CODE REVIEW

File: chaincode/fabcar/java/src/test/java/org/hyperledger/fabric/samples/fabcar/CarQueryResultTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        @Test
+        public void handlesNull() {
+            CarQueryResult cqr = new CarQueryResult("CAR1", new Car("Toyota", "Prius", "blue", "Tomoko"));
+
+            assertThat(cqr).isNotEqualTo(null);
+        }
+    }
+
+    @Test
+    public void toStringIdentifiesCarQueryResult() {
```
Comment (sijocherian) [inline chaincode/fabcar/java/src/test/java/org/hyperledger/fabric/samples/fabcar/CarQueryResultTest.java:87]: is a test on toString() really necessary?  specially asserting on the hash code

toString() is meant for display/logging. If you agree, then let us just remove it from CarTest as well.

File: chaincode/fabcar/java/src/test/java/org/hyperledger/fabric/samples/fabcar/FabCarTest.java [hunk 2]
```diff
@@ -239,7 +239,7 @@ public void whenCarExists() {
 
             Car car = contract.changeCarOwner(ctx, "CAR0", "Dr Evil");
 
-            assertThat(car).isEqualTo(new CarQueryResult("CAR0", new Car("Toyota", "Prius", "blue", "Dr Evil")));
+            assertThat(car).isEqualTo(new Car("Toyota", "Prius", "blue", "Dr Evil"));
```
Comment (sijocherian) [inline chaincode/fabcar/java/src/test/java/org/hyperledger/fabric/samples/fabcar/FabCarTest.java:242]: Thanks for fixing

Rest all Looks good !

File: chaincode/fabcar/java/src/test/java/org/hyperledger/fabric/samples/fabcar/CarQueryResultTest.java (hunk 1 above)
Comment (mbwhite) [reply chaincode/fabcar/java/src/test/java/org/hyperledger/fabric/samples/fabcar/CarQueryResultTest.java:87]: I would disagree; toString() is part of the external API, so it's proper to have tests for it.  

There's nothing worth than debugging a log from a production system only to find you don't have the correct information.

Comment (jt-nti) [reply chaincode/fabcar/java/src/test/java/org/hyperledger/fabric/samples/fabcar/CarQueryResultTest.java:87]: It was also required for the coverage check to pass- it's a small sample so it seems like 100% coverage is reasonable to aspire to

**Pull request merged**